crates/core/src/operations/vacuum.rs

```diff
@@ -235,6 +245,31 @@ impl VacuumBuilder {
             None => Utc::now().timestamp_millis(),
         };
 
+        let keep_files = match &self.keep_versions {
+            Some(versions) => {
+                let mut keep_files: HashSet<String> = HashSet::new();
+                let mut state =
+                    DeltaTableState::try_new(&self.log_store, DeltaTableConfig::default(), Some(0))
```

Comment:
Rather than start from version 0, should we start from `sorted_versions[0]` (with a length check somewhere)? That might avoid the work needed in the first update call if a snapshot can be used.

Reply:
See above, this comes from tracing the logic in load_version.